pkg/problemmetrics/problem_metrics.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package problemmetrics
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/golang/glog"
+
+	"k8s.io/node-problem-detector/pkg/util/metrics"
+)
+
+// GlobalProblemMetricsManager is a singleton of ProblemMetricsManager,
+// which should be used to manage all problem-converted metrics across all
+// problem daemons.
+var GlobalProblemMetricsManager *ProblemMetricsManager
+
+func init() {
+	GlobalProblemMetricsManager = NewProblemMetricsManagerOrDie()
+}
+
+// ProblemMetricsManager manages problem-converted metrics.
+// ProblemMetricsManager is thread-safe.
+type ProblemMetricsManager struct {
+	problemCounter         metrics.Int64MetricInterface
+	problemGauge           metrics.Int64MetricInterface
+	conditionToReason      map[string]string
+	conditionToReasonMutex sync.Mutex
+}
+
+func NewProblemMetricsManagerOrDie() *ProblemMetricsManager {
+	pmm := ProblemMetricsManager{}
+
+	var err error
+	pmm.problemCounter, err = metrics.NewInt64Metric(
+		"problem_counter",
+		"Number of times a specific type of problem have occurred.",
+		"1",
+		metrics.Sum,
+		[]string{"reason"})
+	if err != nil {
+		glog.Fatalf("Failed to create problem_counter metric: %v", err)
+	}
+
+	pmm.problemGauge, err = metrics.NewInt64Metric(
+		"problem_gauge",
+		"Whether a specific type of problem is affecting the node or not.",
+		"1",
+		metrics.LastValue,
+		[]string{"condition", "reason"})
```

Comment:
The `condition` is a k8s specific concept. I think for the problem metrics manager, it is better not to use this term. Call it `type` or something similar? (`type` is consistent with the config)

Reply:
Sounds good. Thanks!